Sources/NIOCore/Channel.swift

```diff
@@ -107,6 +107,11 @@ public protocol Channel: AnyObject, ChannelOutboundInvoker, _NIOPreconcurrencySe
     var allocator: ByteBufferAllocator { get }
 
     /// The `closeFuture` will fire when the `Channel` has been closed.
+    ///
+    /// - Important: This future will never be failed, as it signals when the channel has been closed, and this action cannot fail.
```

Comment:
It can absolutely fail if a `Channel` doesn't implement this as expected, which is quite likely given that this is the first time we're documenting it 😄

I think we should say this it _should_ never be failed.